engine/src/main/java/org/terasology/rendering/dag/StateChange.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+/**
+ *
+ */
+public interface StateChange<T> {
+
+    T getValue();
+}
```

Comment:
I am thinking this interface also needs a second method:

public static StateChange<T> getStateChangeToDefault()

This method should return a StateChange instance that stores the default value for a given opengl parameter. I.e. for the wireframe it would have to store GL_FILL instead of GL_LINE.